airside/camera.py

```
@@ -0,0 +1,206 @@
+"""
+Camera module for handling Picamera2 operations.
+
+This module provides a Camera class that handles Raspberry Pi Camera Module 2
+operations, including initialization and frame capture.
+"""
+
+from picamera2 import Picamera2
+import numpy as np
+import cv2
+import logging
+import time
+
+from util import Colour, Colours
+
+
+class Camera:
+    """
+    Handles Raspberry Pi Camera Module 2 operations
+    """
+
+    # Default camera settings optimized for IR beacon detection
+    DEFAULT_EXPOSURE_TIME = 15  # microseconds - very fast to reduce motion blur
+    DEFAULT_ANALOGUE_GAIN = 16.0  # high gain to amplify IR beacon signal
+    DEFAULT_AUTO_EXPOSURE = False  # disable to maintain consistent brightness
+
+    def __init__(
+        self,
+        camera_index: int = 0,
+        exposure_time: int = DEFAULT_EXPOSURE_TIME,
+        analogue_gain: float = DEFAULT_ANALOGUE_GAIN,
+        auto_exposure: bool = DEFAULT_AUTO_EXPOSURE,
+    ) -> None:
+        """
+        Initialize and configure the Raspberry Pi Camera Module 2.
+
+        Args:
+            camera_index: Camera index (0 for first camera, 1 for second camera)
+            exposure_time: Exposure time in microseconds
+            analogue_gain: Analogue gain value
+            auto_exposure: Enable/disable auto exposure
+        """
+        self.camera_index = camera_index
+        self.exposure_time = exposure_time
+        self.analogue_gain = analogue_gain
+        self.auto_exposure_enabled = auto_exposure
+        self.picam2: Picamera2 | None = None
+
+        # Retry camera initialization
+        while not self._initialize_camera():
+            logging.error(
+                f"Failed to initialize camera {camera_index}, retrying in 1 second..."
+            )
+            time.sleep(1)
```

Silent error condition if cam fails to init permanently. Since this is init is fine to throw some form of hardfault and exit, assuming appropriate metrics emission.